akka-http-core/src/main/scala/akka/http/javadsl/ServerBinding.scala

```diff
@@ -26,5 +30,46 @@ class ServerBinding private[http] (delegate: akka.http.scaladsl.Http.ServerBindi
    *
    * The produced [[java.util.concurrent.CompletionStage]] is fulfilled when the unbinding has been completed.
    */
-  def unbind(): CompletionStage[Done] = delegate.unbind().toJava
+  def unbind(): CompletionStage[Done] =
+    delegate.unbind().toJava
+
+  // TODO docs after review
+  def terminate(hardDeadline: java.time.Duration): CompletionStage[HttpTerminated] = {
+    delegate.terminate(FiniteDuration.apply(hardDeadline.toMillis, TimeUnit.MILLISECONDS))
+      .map(_.asInstanceOf[HttpTerminated])(ExecutionContexts.sameThreadExecutionContext)
+      .toJava
+  }
+
+  /**
+   * Completes when the [[terminate]] is called and server termination is in progress.
+   * Can be useful to make parts of your application aware that termination has been issued,
```

Comment:
Maybe also explain relationship with coordinated shutdown also here?

Reply:
Added it